src/package-resolver.js

```diff
@@ -388,14 +388,18 @@ export default class PackageResolver {
       }
     }
 
+    if (manifest && getExoticResolver(version)) {
+      this.exoticRangeMatch(patterns.map(this.getStrictResolvedPattern.bind(this)), manifest);
```

Comment:
Did you mean to return this? Otherwise I'm a bit confused since we return `null` right after this.